src/importchecker/tests/test_importchecker.py

```diff
@@ -7,6 +7,19 @@
 
 class TestImportChecker(unittest.TestCase):
 
+    def test_no_path_supplied(self):
+        import sys
+
+        output = io.StringIO()
+        orig = sys.argv[:]
+        sys.argv[:] = []
+        with self.assertRaises(SystemExit):
+            main(stdout=output)
+        self.assertEqual(
+            'No path supplied\n',
+            output.getvalue())
+        sys.argv[:] = orig
```

Comment:
If the assertion fails, the cleanup code will not run and sys.argv will remain changed.

(Personally I like `with mock.patch('sys.argv', []): ...`, but if you don't want to add mock to the dependencies, you could write a `self.call_main()` helper that uses try/finally or something like that.)

Reply:
`mock` is ok I guess. Fixed it.